examples/full.rs

```diff
@@ -0,0 +1,416 @@
+                .framerate(30, 1)
+                .enable_picture_type_decision()
+                .encode_config(&mut preset_config.presetCfg),
+        )
+        .expect("Encoder should be initialised correctly");
+
+    // Calculate the number of buffers we need based on the interval of P frames and
+    // the look ahead depth.
+    let num_bufs = usize::try_from(preset_config.presetCfg.frameIntervalP)
+        .expect("frame intervalP should always be positive")
+        + usize::try_from(preset_config.presetCfg.rcParams.lookaheadDepth)
+            .expect("lookahead depth should always be positive");
+
+    let mut output_buffers: Vec<_> = (0..num_bufs)
+        .map(|_| {
+            encoder
+                .create_output_bitstream()
+                .expect("The encoder should be able to create bitstreams")
+        })
+        .collect();
+
+    // Write result to output file "test.bin".
+    let mut out_file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open("test.bin")
+        .expect("Permissions and available space should allow creating a new file");
+
+    // Create Vulkan buffers
+    // 4.1.2. Input buffers allocated externally
+
+    // Generate each of the frames.
+    let file_descriptors = (0..FRAMES)
+        .map(|f| {
+            create_buffer(
+                vulkan_device.clone(),
+                memory_type_index,
+                WIDTH,
+                HEIGHT,
+                f,
+                FRAMES,
+            )
+        })
+        .collect::<Vec<_>>();
+
+    // Encode each of the frames.
+    for (i, file_descriptor) in file_descriptors.into_iter().enumerate() {
+        let output_buffer = &mut output_buffers[i % num_bufs];
+
+        let mut input_buffer =
+            fd_into_nvenc_resource(&encoder, buffer_format, WIDTH, HEIGHT, file_descriptor);
+
+        encoder
+            .encode_picture(NV_ENC_PIC_PARAMS::new(
+                WIDTH,
+                HEIGHT,
+                &mut input_buffer,
+                output_buffer,
+                buffer_format,
+                NV_ENC_PIC_STRUCT::NV_ENC_PIC_STRUCT_FRAME,
+            ))
+            .expect("Encoder should be able to encode valid pictures");
+
+        let out = output_buffer
+            .lock_and_read()
+            .expect("Buffer should be fully filled and not locked");
+        out_file
+            .write_all(out)
+            .expect("Writing should succeed because `out_file` was opened with write permissions");
+    }
+
+    // Notifying the End of Input Stream with end of stream picture.
+    // Note that output is still generated here.
+    let output_buffer = &mut output_buffers[0];
+    encoder
+        .encode_picture(NV_ENC_PIC_PARAMS::end_of_stream())
+        .expect("Should be able to encode empty end of stream file");
+
+    let out = output_buffer
+        .lock_and_read()
+        .expect("Buffer should be fully filled and not locked");
```

Comment:
You should either pass in the output buffer or not read from it at all. At the moment you are re-reading old output.

Reply:
Not reading from it at all is correct, I think. EOS should produce any output I would think  (see also a few comments above).